### PR TITLE
Make data explorer non-blocking with docker compose

### PR DESCRIPTION
**Pull request by PhilippeMoussalli**
Small PR to avoid showing the localhost url before the image is build

### CODE REVIEW

**Comment by RobbeSneyders** [comment]
The `docker run` call is blocking, so with this change the url will only be shown after you stop the explorer again. I think the only way to improve on the current situation is by making the `docker run` command non-blocking, which would probably be an improvement for usage in the notebooks anyway. We might want to use docker compose then instead of docker so the explorer is only started once and easy to bring down.

**Comment by PhilippeMoussalli** [comment]
> We might want to use docker compose then instead of docker so the explorer is only started once and easy to bring down.

Makes sense indeed, it's always a bit clunky to stop it from the notebook. I'll update the PR

**Comment by PhilippeMoussalli** [comment]
> > We might want to use docker compose then instead of docker so the explorer is only started once and easy to bring down.
> 
> Makes sense indeed, it's always a bit clunky to stop it from the notebook. I'll update the PR

Updated the PR, the explorer now uses docker compose to launch the service, slight change to the cli command: 

starting the service:

```
fondant explore start -b <base_path>
```

Stopping the service: 

```
fondant explore stop
```

**Review by RobbeSneyders — COMMENTED** [review]
Thanks @PhilippeMoussalli!

I had it a bit different in my mind, thinking about a docker compose file which we include in the Fondant package. But this approach indeed makes more sense so the compose file can be generated with different options :+1: 

The approach and code looks good to me, but I still want to test it. Will probably do so tomorrow!

File: src/fondant/explore.py [hunk 1]
```diff
@@ -97,27 +76,106 @@ def run_explorer_app(  # type: ignore
...
+        output_path,
+        "up",
+        "--build",
+        "--pull",
+        "always",
+        "--remove-orphans",
+        "--detach",
+    ]
+
+    subprocess.call(cmd, stdout=subprocess.PIPE)  # nosec
```
Comment (RobbeSneyders) [inline src/fondant/explore.py:151]: I think we should use [`check_call`](https://docs.python.org/3/library/subprocess.html#subprocess.check_call) here so we don't print the messages below if this fails (eg. when the docker daemon is not running).

**Review by RobbeSneyders — COMMENTED** [review]
Just tested @PhilippeMoussalli. Works well, I would just catch errors with the docker command, see my comment below.

File: src/fondant/explore.py [hunk 2]
```diff
@@ -148,7 +148,10 @@ def run_explorer_app(  # type: ignore  # noqa: PLR0913
         "--detach",
     ]
 
-    subprocess.call(cmd, stdout=subprocess.PIPE)  # nosec
+    try:
+        subprocess.check_call(cmd, stdout=subprocess.PIPE)  # nosec
+    except subprocess.CalledProcessError as e:
+        raise OSError(e.returncode)
```
Comment (RobbeSneyders) [inline src/fondant/explore.py:154]: Why did you choose an `OSError` here? I would probably have gone for a `SystemExit` exception myself.

Comment (PhilippeMoussalli) [reply src/fondant/explore.py:154]: Might make more sense here indeed, updated

**Review by RobbeSneyders — APPROVED** [review]
Thanks @PhilippeMoussalli!

**Pull request merged**